src/components/comparativeD3Graph.js

```diff
@@ -0,0 +1,269 @@
+            } else {
+              historyTemperature = Number(this.props.historyData.tomorrowInHistory.history.observations[k].tempi)
+            }
+            if (historyTemperature < 0 && index >= 1) {
+              historyTemperature = Number(dataForRender[index-1].historyTemperature)
+            }
+            else if (historyTemperature < 0 && index === 0) {
+              historyTemperature = Math.round((Number((this.props.historyData.tomorrowInHistory.history.dailysummary[0].maxtempi)) + Number((this.props.historyData.tomorrowInHistory.history.dailysummary[0].mintempi))) / 2)
+            }
+
+            dataForRender.push({
+              'todaysTemperature': Number(this.props.currentData.hourly_forecast[index].temp.english),
+              'historyTemperature': historyTemperature,
+              'time': index
+            })
+            k++
+          }          
+        }
+        console.log(9, dataForRender)
+        //
+
+        const vis = d3.select('#visulization')
+        vis.selectAll('*').remove()
+        
+        /*
+        // stand-in x axis
+        const standinXaxis = d3.axisBottom().scale(xScale).ticks(7)
+        vis.append("svg:g")
+            .attr("class", "x axis")
+            .attr("transform", "translate(50, 150)")
+            .call(standinXaxis)
+        //
+        */
+
+        //draw x axis
+        const xScale = d3.scaleLinear().range([-20, WIDTH-50]).domain([dataForRender[0].time, dataForRender[23].time])
+        const d3ParseScale = d3.scaleTime()
+              .domain([d3.timeParse('%Y-%m-%dT%H:%M:%S')(today), d3.timeParse('%Y-%m-%dT%H:%M:%S')(tomorrow)])
+              .range([30, WIDTH]);
+        const drawScale = function(scale, dst) {
+            const xAxis = d3.axisBottom(scale);
+            vis.append('g')
+              .attr('class', 'axisLine')
+              .attr("transform", "translate(0, 250)")
+              .attr('stroke', 'white')
+              .attr('stroke-width', 1)
+              .call(xAxis)
+        }
+        drawScale(d3ParseScale,[]);
+        //
+
+
+        //get lowest and highest temperature
+        const minTemperature = Math.min(d3.min(dataForRender, function(d) { return d.todaysTemperature }), d3.min(dataForRender, function(d) { return d.historyTemperature }))
+        const maxTemperature = Math.max(d3.max(dataForRender, function(d) { return d.todaysTemperature }), d3.max(dataForRender, function(d) { return d.historyTemperature }))
+
+        //draw y axis
+        const yScale = d3.scaleLinear().range([250, 20]).domain([minTemperature - 2, maxTemperature + 2])
+        const yAxis = d3.axisLeft().scale(yScale)
+        vis.append("g")
+              .attr('class', 'axisLine')
+              .attr("transform", "translate(30, 0)")
+              .attr('stroke', 'white')
+              .attr('stroke-width', 1)
+              .call(yAxis)
+
+        
+        //create path drawers      
+        const lineGen_current = d3.line()
```

Comment:
A lot of this D3 code looks copy-pasted.

I can tell by:
- the spacing/indentation not matching the overall indentation/spacing patterns of your own code
- `function(d) { return xScale(d.time); }` doesn't look like your code.